chain/bitcoind_client.go

```diff
@@ -1097,6 +1024,9 @@ func (c *BitcoindClient) filterTx(tx *wire.MsgTx,
 	blockDetails *btcjson.BlockDetails, notify bool) (bool,
 	*wtxmgr.TxRecord, error) {
 
+	c.mu.Lock()
```

Comment:
nit: move the locking down to where it is first needed

Reply:
Fixed.